src/SkeletalTrapezoidation.cpp

```diff
@@ -0,0 +1,3083 @@
+            ending_edge_is_set_before_start = !after_start;
+            ending_vd_edge = edge;
+            check_secondary_edge = false;
+        }
+        if (false && check_secondary_edge && edge->is_secondary()
+               != LinearAlg2D::pointLiesOnTheRightOfLine(v1, from, to)
+            &&    LinearAlg2D::pointLiesOnTheRightOfLine(v0, from, to)
+            && (LinearAlg2D::getDist2FromLineSegment(v0, from, v1) <= 5 // TODO: magic value
+                || LinearAlg2D::getDist2FromLineSegment(v0, to, v1) <= 5)
+        )
+        { // edge crosses source segment
+            // TODO: handle the case where two consecutive line segments are collinear!
+            // that's the only case where a voronoi segment doesn't end in a polygon vertex, but goes though it
+            if (LinearAlg2D::pointLiesOnTheRightOfLine(VoronoiUtils::p(edge->vertex1()), source_segment.from(), source_segment.to()))
+            {
+                ending_vd_edge = edge;
+            }
+            else
+            {
+                starting_vd_edge = edge;
+            }
+            first = false;
+            continue;
+        }
+        first = false;
+    }
+    
+    assert(starting_vd_edge && ending_vd_edge);
+    assert(starting_vd_edge != ending_vd_edge);
+    
+    start_source_point = source_segment.to();
+    end_source_point = source_segment.from();
+}
+
+SkeletalTrapezoidation::SkeletalTrapezoidation(const Polygons& polys, float transitioning_angle
+, coord_t discretization_step_size
+, coord_t transition_filter_dist
+, coord_t beading_propagation_transition_dist
+)
+: polys(polys)
+, transitioning_angle(transitioning_angle)
+, discretization_step_size(discretization_step_size)
+, transition_filter_dist(transition_filter_dist)
+, beading_propagation_transition_dist(beading_propagation_transition_dist)
+{
+    init();
+}
+
+void SkeletalTrapezoidation::init()
+{
+    std::vector<Point> points; // remains empty
+
+    std::vector<Segment> segments;
+    for (size_t poly_idx = 0; poly_idx < polys.size(); poly_idx++)
+    {
+        ConstPolygonRef poly = polys[poly_idx];
+        for (size_t point_idx = 0; point_idx < poly.size(); point_idx++)
+        {
+            segments.emplace_back(&polys, poly_idx, point_idx);
+        }
+    }
+
+    vd_t vd;
+    construct_voronoi(points.begin(), points.end(),
+                                        segments.begin(), segments.end(),
+                                        &vd);
+
+#ifdef DEBUG
+    VoronoiUtils::debugOutput("output/vd.svg", vd, points, segments);
+#endif
+
+    for (const vd_t::edge_type& edge : vd.edges())
+    {
+        assert(edge.vertex0() == edge.twin()->vertex1());
+        assert(edge.vertex1() == edge.twin()->vertex0());
+        assert(edge.vertex1() == edge.next()->vertex0());
+        assert(edge.vertex0() == edge.prev()->vertex1());
+    }
+
+    
+    for (vd_t::cell_type cell : vd.cells())
+    {
+        if (!cell.incident_edge())
+        { // there is no spoon
```

Comment:
From this I'm understanding that instead of bending this incident edge, we're going to bend the world!